File: howto/letsencrypt/goweb/serve-https.go
package main

import (
	"log"
	"net/http"
)

func main() {
	fs := http.FileServer(http.Dir("."))
	http.Handle("/", fs)

	log.Println("Serving on https://localhost:4343...")
	err := http.ListenAndServeTLS(":4343", "../certs/fullchain.pem", "../certs/privkey.pem", nil)
	if err != nil {
		log.Fatal(err)
	}
}

File: howto/letsencrypt/pyweb/serve-https.py
#!/usr/bin/env python3

import http.server
import ssl
import os

port = 4343
server_address = ('localhost', port)
httpd = http.server.HTTPServer(server_address, http.server.SimpleHTTPRequestHandler)

ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
ssl_context.load_cert_chain('../certs/fullchain.pem', '../certs/privkey.pem')

httpd.socket = ssl_context.wrap_socket(httpd.socket, server_side=True)

cwd = os.getcwd()
print('----------------------------------------------')
print(f"HTTPS server running on port {port}")
print(f"Serving files from {cwd}")
print('----------------------------------------------')

httpd.serve_forever()
